### PR TITLE
Improvements "Callback" rule

### PR DESCRIPTION
**Pull request by thenotsoft**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  |

### CODE REVIEW

File: src/Rule/Callback.php [hunk 1]
```diff
@@ -12,24 +12,61 @@ class Callback extends Rule
 {
     private $callback;
 
-    public function __construct(callable $callback)
+    public function __construct($callback)
     {
+        if (!(is_callable($callback) || (is_array($callback) && count($callback) === 2))) {
```
Comment (xepozz) [inline src/Rule/Callback.php]: Why did you do it?
http://sandbox.onlinephpfunctions.com/code/13996f1b47b5103604d7f4263786eb240a31f996

Comment (thenotsoft) [reply src/Rule/Callback.php]: you're right, `is_callable` will be enough

Comment (xepozz) [reply src/Rule/Callback.php]: to use private method we can pass the closure like the following:
```php
class Form 
{
    public function validate()
    {
        $rule = new Callback(function ($value) {
            $this->checkValue($value);
        });
        // or shortly
        $rule = new Callback(fn ($value) => $this->checkValue($value));
    }

    private function checkValue(string $value)
    {
        // some check
    }
}
```

**Comment by samdark** [comment]
Thanks!

**Comment by samdark** [comment]
Revered because of https://github.com/yiisoft/validator/pull/58#discussion_r386103133 and potential issues of calling private methods on foreign classes. https://github.com/yiisoft/validator/commit/ddc766d40bd55ca16dbe4620f028daa527588e80

**Pull request merged**